lib/absinthe/schema/notation.ex

```diff
@@ -1866,6 +1905,18 @@ defmodule Absinthe.Schema.Notation do
 
   defp expand_ast(ast, env) do
     Macro.prewalk(ast, fn
+      # We don't want to expand `@bla` into `Module.get_attribute(module, @bla)` because this
+      # function call will fail if the module is already compiled. Remember that the ast gets put
+      # into a generated `__absinthe_blueprint__` function which is called at "__after_compile__"
+      # time. This will be after a module has been compiled if there are multiple modules in the
+      # schema (in the case of an `import_types`).
+      #
+      # Also see test "test/absinthe/type/import_types_test.exs"
+      # "__absinthe_blueprint__ is callable at runtime even if there is a module attribute"
+      # and it's comment for more information
+      {:@, _, _} = node ->
```

Comment:
I haven't quite worked out why this work, because I was sure this would fail with:

```
defmodule Foo do
  use Absinthe.Schema
  
  query do
  end
  @msg "It's a User"

  @desc @msg
  object :user do
  end

  @msg "It's a dog"

  @desc @msg
  object :dog do
  end
end
```

However based on my local testing:
```
iex(4)> Foo.__absinthe_type__(:user).description
"It's a User"
iex(5)> Foo.__absinthe_type__(:dog).description 
"It's a dog"
```

Works great!

Reply:
The value assigned to `@desc` module attribute is evaluated outside of Absinthe, so actually the example you showed there works without any magic (aside from elixir itself)